app/client/keybase/doc/vault.md

```diff
@@ -0,0 +1,225 @@
+    --vault-addr http://127.0.0.1:8200/ \
+    addr_goes_here
+
+# {"level":"info","address":"new_addr_goes_here","parent":"addr_goes_here","index":0,"stored":true,"time":"2023-02-28T09:26:11-04:00","message":"Child key derived"}
+
+p1 Keys Delete --keybase vault \
+    --vault-token dev-only-token \
+    --vault-addr http://127.0.0.1:8200/ \
+    addr_goes_here
+
+## TODO: add SignTx and VerifyTx
+
+p1 Keys SignTx --keybase vault \
+    --vault-token dev-only-token \
+    --vault-addr http://127.0.0.1:8200/
+p1 Keys VerifyTx --keybase vault \
+    --vault-token dev-only-token \
+    --vault-addr http://127.0.0.1:8200/
+
+```
+
+### Example using config.json
+
+```sh
+cat << EOF > config-with-vault.json
+{
+  "keybase": {
+    "type": "vault",
+    "vault_addr": "http://127.0.0.1:8200",
+    "vault_token": "dev-only-token",
+    "vault_mount": "secret"
+  }
+}
+EOF
+
+
+p1 Keys Create --config config-with-vault.json
+
+# note the address
+# {"level":"info","address":"addr_goes_here","time":"2023-02-24T23:12:06-04:00","message":"New Key Created"}
+
+p1 Keys Get --config config-with-vault.json
+
+# {"level":"info","address":"addr_goes_here","public_key":"d82cd23f4809491c04ab456dd9714e647093bcc6cb649a8510f4d54c194f80ea","time":"2023-02-24T23:14:01-04:00","message":"Found key"}
+
+p1 Keys Export addr_goes_here --config config-with-vault.json
+
+## note the private key
+# {"level":"info","private_key":"{\"kdf\":\"scrypt\",\"salt\":\"salty_goes_ere\",\"secparam\":\"12\",\"hint\":\"\",\"ciphertext\":\"ciphertext_goes_here\"}","time":"2023-02-24T23:12:53-04:00","message":"Key exported"}
+
+p1 Keys Import --config config-with-vault.json \
+    --import_format json "{\"kdf\":\"scrypt\",\"salt\":\"salty_goes_ere\",\"secparam\":\"12\",\"hint\":\"\",\"ciphertext\":\"ciphertext_goes_here\"}"
+
+p1 Keys List --config config-with-vault.json
+
+# {"level":"info","addresses":["addr_goes_here"],"time":"2023-02-24T23:14:44-04:00","message":"Get all keys"}
+
+p1 Keys Sign --config config-with-vault.json \
+    addr_goes_here abcd
+
+# note the signature
+# {"level":"info","signature":"signature_goes_here","address":"addr_goes_here","time":"2023-02-24T23:15:18-04:00","message":"Message signed"}
+
+p1 Keys Verify --config config-with-vault.json \
+    addr_goes_here abcd signature_goes_here
+
+# {"level":"info","address":"addr_goes_here","valid":true,"time":"2023-02-24T23:16:05-04:00","message":"Signature checked"}
+
+p1 Keys Update --config config-with-vault.json addr_goes_here
+
+
+p1 Keys Keys DeriveChild addr_goes_here 0 --config config-with-vault.json \
+    addr_goes_here
+
+# {"level":"info","address":"new_addr_goes_here","parent":"addr_goes_here","index":0,"stored":true,"time":"2023-02-28T09:26:11-04:00","message":"Child key derived"}
+
+p1 Keys Delete --config config-with-vault.json addr_goes_here
+
+## TODO: add SignTx and VerifyTx
+
+p1 Keys SignTx --config config-with-vault.json
+p1 Keys VerifyTx --config config-with-vault.json
+
+```
```

Comment:
Why don't I see any secrets even though I can export?

![Screenshot 2023-03-24 at 4 00 02 PM](https://user-images.githubusercontent.com/1892194/227658644-1a3614d3-d0cc-400d-b4ad-d8354085c632.png)



![Screenshot 2023-03-24 at 3 58 50 PM](https://user-images.githubusercontent.com/1892194/227658529-24f1bee7-8e1c-4bb3-9408-9a3287386f9c.png)

Reply:
ah this was a great catch, ty. fixed here https://github.com/pokt-network/pocket/pull/537/commits/cb796ee4cf74e4d713850c48e6be244e685a7199

Reply:
1. I rebuilt the binary
2. Created a new key
3. Launched vault - still don't see anything
4. What do I need to do?

![Screenshot 2023-03-27 at 1 51 12 PM](https://user-images.githubusercontent.com/1892194/228063729-1e423ecc-4403-4326-8b34-66eda1febd08.png)

Reply:
hmmm, not sure @Olshansk 

- how did u attempt to connect to the vault this time? flags, config, or env variables?
- what happens when you run `p1 --keybase vault keys list` without configuring address or token or starting the vault? Do you see `{"level":"fatal","error":"Get \"https://127.0.0.1:8200/v1/secret/metadata?list=true\": dial tcp 127.0.0.1:8200: connect: connection refused","time":"2023-03-27T17:06:58-04:00","message":"failed to execute command"}`
- where is ur vault hosted? If docker, are the ports published?

Reply:
https://user-images.githubusercontent.com/1892194/228078783-2fffcd3c-89a3-4b60-92a3-56e686d0bcad.mov

Reply:
Apologies, since no config file was provided, it returned early and did not go through the normal viper unmarshal flow., fixed in https://github.com/pokt-network/pocket/pull/537/commits/0382fa25c85b64368b0185e7b970f651630f3b47

Reply:
![Screenshot 2023-03-28 at 1 45 23 PM](https://user-images.githubusercontent.com/1892194/228362648-18e2226c-8de2-454d-86ba-f54ec12b8894.png)
![Screenshot 2023-03-28 at 1 45 19 PM](https://user-images.githubusercontent.com/1892194/228362655-8d108d62-a57a-4c56-bc02-e975d0c7226f.png)